_posts/2024-08-25-mot-tracking-metrics.md

```diff
@@ -0,0 +1,172 @@
+
+## IDF1
+
+IDF1 addresses some of MOTA's limitations by focusing on how long the tracker correctly identifies an object, rather than just counting errors. It's based on the concept of Identification Precision (IDP) and Identification Recall (IDR).
+
+It computes the assignment between prediction and ground truth objects across the entire video, rather than frame by frame. 
+
+The metric is simple:
+
+$$
+\text{IDF1} = \frac{2 * \text{IDTP}}{2 * \text{IDTP} + \text{IDFP} + \text{IDFN}}
+$$
+
+Where:
+
+- **IDTP** (ID True Positive): The number of correctly identified detections
+- **IDFP** (ID False Positive): Tracker hypotheses that don't match any ground truth
+- **IDFN** (ID False Negative): Ground truth trajectories that aren't tracked
+
+The global assignment is computed using the Hungarian algorithm. It pick the best combination between prediction and ground truth for the whole video. It is easier to understand by observing the image introduced in HOTA paper:
+
+
+<div class="post-center-image" style="max-width: 800px; margin: 0 auto;">
+    {% picture pimage /assets/images/fullsize/posts/2024-08-25-mot-tracking-metrics/idf1.png --alt IDF1 metric diagram %}
+</div>
+
+{:refdef: class="image-caption"}
+*Figure 3. IDF1 metric diagram*
+{: refdef}
+
+The main problem I see with IDF1 is finding the best one-to-one matching between predicted and ground truth trajectories for the entire sequence. However, this approach can sometimes oversimplify complex tracking scenarios:
+
+Imagine a corner kick in football. A tracker might correctly follow Player A running into the box, lose them in a cluster, and then mistakenly pick up Player B after the ball is cleared. IDF1 might treat this as one partially correct track for either Player A or B, ignoring that it's correct for different players at different times.
+
+This simplification can misrepresent a tracker's performance in complex situations like crowded football plays, where player interactions and occlusions are frequent.
+
+Key advantages of IDF1:
+
+1. It's more sensitive to **long-term tracking consistency**.
+2. It balances **precision and recall** of identity predictions.
+3. It's less affected by the number of objects in the scene than MOTA.
+
+However, IDF1 also has limitations:
+
+1. **IDF1 can decrease when improving detection**. Just avoiding FP can result in a better metric (A vs C)
+2. **IoU is fixed** so more or less detection accuracy is not reflected on the metric
+
+More limitations are presented in the HOTA paper. I recommend you to have a read because it is very well explained and intuitive.
+
+## HOTA (Higher Order Tracking Accuracy)
+
+HOTA is a more recent metric designed to address the limitations of both MOTA and IDF1. It aims to provide a balanced assessment of detection and association performance. HOTA can be broken down into DetA (Detection Accuracy) and AssA (Association Accuracy), allowing separate analyses of these aspects.
+
+The core HOTA formula is:
+
+$$
```

Comment:
Given AssA is used here, I would emphasize the need to define it properly in the intro section